### PR TITLE
minor: various Docker optimizations

### PR DESCRIPTION
**Pull request by dunglas**
* make `bin/setup-runtime` compatible with Alpine Linux
* read the Dockerfile from stdin: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#pipe-dockerfile-through-stdin
* optimize Composer autoloader
* disable the cache of `apk`

### CODE REVIEW

File: bin/setup-runtime [hunk 1]
```diff
@@ -52,6 +52,8 @@ china)
 
 esac
 
+apk add --no-cache curl || true
```
Comment (crazywhalecc) [inline bin/setup-runtime:55]: It's working on Alpine, but other distros will show a warning message.

**Review by crazywhalecc — APPROVED** [review]
Checked, it works well (macOS, Debian, Alpine)

File: bin/setup-runtime (hunk 1 above)
Comment (dunglas) [reply bin/setup-runtime]: warning hided

**Comment by dunglas** [comment]
By the way, there is something weird on Mac M1: the Docker image is named `cwcc-spc-x86_64` while actually built for the aarch64 platform. Now that Docker (and Alpine containers) natively supports [multi-platform images](https://docs.docker.com/build/building/multi-platform/), I setting the `--platform` flag instead of changing the `FROM` clause should work.

**Comment by crazywhalecc** [comment]
Oh I found that GH Action build failed. https://github.com/crazywhalecc/static-php-cli/actions/runs/5945970161/job/16125847564

Is that related to this docker optimization?

**Pull request merged**